pkg/pillar/cmd/baseosmgr/context.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2022 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package baseosmgr
+
+import (
+	"flag"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/agentbase"
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/lf-edge/eve/pkg/pillar/pubsub"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/worker"
+	"github.com/sirupsen/logrus"
+)
+
+type baseOsMgrContext struct {
+	agentBaseContext     *agentbase.Context
+	pubBaseOsStatus      pubsub.Publication
+	pubContentTreeConfig pubsub.Publication
+	pubZbootStatus       pubsub.Publication
+	pubBaseOsMgrStatus   pubsub.Publication
+
+	subGlobalConfig      pubsub.Subscription
+	globalConfig         *types.ConfigItemValueMap
+	GCInitialized        bool
+	subBaseOsConfig      pubsub.Subscription
+	subBaseOs            pubsub.Subscription
+	subZbootConfig       pubsub.Subscription
+	subContentTreeStatus pubsub.Subscription
+	subNodeAgentStatus   pubsub.Subscription
+	subZedAgentStatus    pubsub.Subscription
+	rebootReason         string    // From last reboot
+	rebootTime           time.Time // From last reboot
+	rebootImage          string    // Image from which the last reboot happened
+	currentUpdateRetry   uint32    // UpdateRetryCounter from last retry; it will be sent for info
+	configUpdateRetry    uint32    // UpdateRetryCounter from config; to avoid loop after reboot with failed testing
+
+	baseOsConfigRestarted bool
+
+	worker worker.Worker // For background work
+	// cli options
+	versionPtr *bool
+}
+
+func newBaseOsMgrContext(_ *pubsub.PubSub, logger *logrus.Logger, log *base.LogObject, arguments []string) *baseOsMgrContext {
+	ctx := baseOsMgrContext{}
+
+	// load saved values or fill with 0
+	ctx.currentUpdateRetry = readSavedCurrentRetryUpdateCounter()
+	ctx.configUpdateRetry = readSavedConfigRetryUpdateCounter()
```

Comment:
It is quite confusing to have these two lines in something which is just supposed to initialize the context/state.
Even though it doesn't do side-effect modifications the fact that it reads some state from /persist means that it should be visible in the overall flow and not hidden here.

Reply:
Let's not split the logic in the PR, modified.